python/pylibraft/pylibraft/common/ai_wrapper.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import numpy as np
+
+from pylibraft.common import input_validation
+
+
+class ai_wrapper:
```

Comment:
Could we have the `cai_wrapper` extend the `ai_wrapper`? This way we can define the common methods once and just have the `cai_wrapper` implement the `copy_to_host()` (and maybe the `ai_wrapper` could implement a `copy_to_device()`?)

Reply:
Changed `cai_wrapper` to be a child class of `ai_wrapper`. 

The `copy_to_host` method is only part of  `device_ndarray`, but not `cai_wrapper`. As I understand the `[c]ai_wrapper` classes just provide convenience for inspecting the array properties, but they do not need to provide copy operation for the data.

Reply:
Ah yes, my mistake. I guess then the only difference really is the name of array interface that's being called so even better to have it this way.